packages/core/src/domain/configuration/experimentalFeatures.ts

```diff
@@ -0,0 +1,26 @@
+let enabledExperimentalFeatures: Set<string>
+
+export function updateExperimentalFeatures(enabledFeatures: string[]): void {
+  // Safely handle external data
+  if (!Array.isArray(enabledFeatures)) {
+    return
+  }
+
+  if (!enabledExperimentalFeatures) {
+    enabledExperimentalFeatures = new Set(enabledFeatures)
+  }
```

Comment:
no unit test for `!enabledExperimentalFeatures`, what was your intent?

Reply:
You cannot add flags to a set if the set does not exist. This logic is tested via lines 37,39,40.